src/Module/ActivityPub/Featured.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\ActivityPub;
+
+use Friendica\BaseModule;
+use Friendica\Core\System;
+use Friendica\Model\User;
+use Friendica\Protocol\ActivityPub;
+
+/**
+ * ActivityPub featured posts
+ */
+class Featured extends BaseModule
+{
+	protected function rawContent(array $request = [])
+	{
+		if (empty($this->parameters['nickname'])) {
+			throw new \Friendica\Network\HTTPException\NotFoundException();
+		}
+
+		$owner = User::getOwnerDataByNick($this->parameters['nickname']);
+		if (empty($owner)) {
+			throw new \Friendica\Network\HTTPException\NotFoundException();
+		}
+
+		$page = $_REQUEST['page'] ?? null;
+
+		$outbox = ActivityPub\Transmitter::getFeatured($owner, $page);
+		System::jsonExit($outbox, 'application/activity+json');
```

Comment:
Please use the `$this->response` class instead.

I try to avoid `exit` during calls because they are hard to catch and impossible to test

Reply:
Is there some general function that does all the trick in a single call? Otherwise I suggest that we at first have got a look into the code for all occurrences where we simply `echo` out the content and then `exit`. These can - in a first step - be replaced with the system function. In a next step this function can then be replaced with a functionality that does the whole job.

Reply:
Yep, good idea and you already replaced it :)
Step by step :-)